flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -582,7 +583,18 @@ private void startTriggeringCheckpoint(CheckpointTriggerRequest request) {
 
 							return null;
 						},
-						timer));
+						timer)
+					.exceptionally(error -> {
+						isTriggering = false;
+						if (!isShutdown()) {
+							ExceptionUtils.rethrow(error);
```

Comment:
```suggestion
							throw new CompletionException(error)
```

is better because it won't extend the stack trace with another exception.